### PR TITLE
build: add build script for PowerShell

### PR DESCRIPTION
**Pull request by bobotu**
### What problem does this PR solve?
For Windows users, PowerShell is easier to set up than Cygwin, etc.

Although some unit tests cannot be run on Windows yet (#19088), the ability to run `make check` directly on Windows is still handy for developers on the Windows platform.

Because PowerShell 7 is a cross-platform shell, you can also run this build script on Linux and macOS. I have tested it on openSUSE with PowerShell 7.0.3, this means in theory it can be invoked as a PowerShell remote command.

To use this new build script, you should [install PowerShell 7](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell?view=powershell-7).

The build script is based on `InvokeBuild` module, you can install it by
```powershell
Install-Module InvokeBuild
```

Now you can build TiDB by run

```powershell
Invoke-Build
```

You can use `Invoke-Build ?` to list all supported tasks
```
Name                Jobs                                                Synopsis
----                ----                                                --------
BuildFailPoint      {}
EnableFailPoint     {BuildFailPoint, {}}
DisableFailPoint    {BuildFailPoint, {}}
BuildErrcheck       {}
RunErrcheck         {BuildErrcheck, {}}
BuildRevive         {}
RunRevive           {BuildRevive, {}}
BuildUnconvert      {}
RunUnconvert        {BuildUnconvert, {}}
BuildStaticCheck    {}
RunStaticCheck      {BuildStaticCheck, {}}
DownloadLinter      {}
RunLinter           {DownloadLinter, {}}
GoModTidy           {}
CheckTestSuite      {}
RunGoVet            {}
FormatCode          {}
Check               {FormatCode, RunErrCheck, RunUnconvert, RunRevive…} Check code quality with some analyzers.
Build               {}                                                  Build TiDB server.
BuildExplainTest    {}
ExplainTest         {Build, BuildExplainTest, {}}                       Run explain tests.
CheckDep            {}                                                  Check dependency.
GoTest              {BuildFailPoint, {}}                                Run unit tests.
GoRaceTest          {BuildFailPoint, {}}                                Run tests with race detecter enabled.
GoLeakTest          {BuildFailPoint, {}}                                Run tests with leak checker enabled.
TiKVIntegrationTest {BuildFailPoint, {}}                                Run some tests with real TiKV.
GoGenerate          {}                                                  Ensure generated code is up to date.
Test                {ExplainTest, CheckDep, GoTest, GoGenerate}         Run common tests.
Dev                 {Check, Test}                                       Check and Test.
.                   Build                                               Build TiDB server.
```

To make a long story short, the commonly used `make`, `make check`, `make gotest`, `make explaintest`, `make test`, and `make dev` is included in PowerShell version build script.

You can set an alias for `Invoke-Build`, and because PowerShell is case insensitive, you can use it in this way.
```powershell
Set-Alias make Invoke-Build

make check
make gotest
make dev
make explaintest
```

### Release note 

-  No release note

### CODE REVIEW

File: .build.ps1 [hunk 1]
```diff
@@ -0,0 +1,370 @@
+# Copyright 2020 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
```
Comment (jackysp) [inline .build.ps1:7]:
```suggestion
#     http://www.apache.org/licenses/LICENSE-2.0
```

**Comment by tiancaiamao** [comment]
LGTM

I expect the code to work on more platforms although I have not been using Windows for a while.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**